ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoTransactionTreesSpec.scala

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.store.dao
+
+import com.digitalasset.daml.lf.transaction.Node.{NodeCreate, NodeExercises}
+import com.digitalasset.daml.lf.value.Value.AbsoluteContractId
+import com.digitalasset.ledger.EventId
+import com.digitalasset.platform.ApiOffset
+import org.scalatest._
+
+private[dao] trait JdbcLedgerDaoTransactionTreesSpec
+    extends OptionValues
+    with Inside
+    with LoneElement {
+  this: AsyncFlatSpec with Matchers with JdbcLedgerDaoSuite =>
+
+  behavior of "JdbcLedgerDao (lookupTransactionTreeById)"
+
+  it should "return nothing for a mismatching transaction id" in {
+    for {
+      (_, tx) <- storeCreateTransaction()
+      result <- ledgerDao.transactionsReader
+        .lookupTransactionTreeById(transactionId = "WRONG", Set(tx.submittingParty.get))
+    } yield {
+      result shouldBe None
+    }
+  }
+
+  it should "return nothing for a mismatching party" in {
+    for {
+      (_, tx) <- storeCreateTransaction()
+      result <- ledgerDao.transactionsReader
+        .lookupTransactionTreeById(tx.transactionId, Set("WRONG"))
+    } yield {
+      result shouldBe None
+    }
+  }
+
+  it should "return the expected flat transaction for a correct request (create)" in {
```

Comment:
This test reads the transaction tree, not the flat transaction.

Reply:
Yes, thanks, copy-pasta.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/5089/commits/eba57bf61d9a7587aa53a066fc83b479c6980983